src/region/RegionAnnotations.tsx

```diff
@@ -38,6 +44,32 @@ export default class RegionAnnotations extends React.PureComponent<Props, State>
 
     state: State = {};
 
+    componentDidUpdate(_prevProps: Props, prevState: State): void {
+        const { popupRef: prevPopupRef } = prevState;
+        const { popupRef } = this.state;
+
+        if (prevPopupRef && prevPopupRef !== popupRef) {
+            prevPopupRef.removeEventListener('mousedown', handleEvent);
```

Comment:
Should this be handled by the popup itself, instead?

Reply:
Good point, I've moved that logic into the PopupReply and that reduced the amount of code churn in consuming components